check port status and skip fdb add if port is not in learning or forwading state

diff --git a/net/bridge/br_fdb.c b/net/bridge/br_fdb.c
index f01ff2e..038ef86 100644
--- a/net/bridge/br_fdb.c
+++ b/net/bridge/br_fdb.c
@@ -683,6 +683,12 @@ static int fdb_add_entry(struct net_bridge_port *source, const __u8 *addr,
 	struct net_bridge_fdb_entry *fdb;
 	bool modified = false;
 
+	/* skip add if the port is not up*/
+	if (!(state & NUD_PERMANENT || state & NUD_NOARP) &&
+		!(source->state == BR_STATE_LEARNING
+			|| source->state == BR_STATE_FORWARDING))
+		return 0;
+
 	fdb = fdb_find(head, addr, vid);
 	if (fdb == NULL) {
 		if (!(flags & NLM_F_CREATE))
